### PR TITLE
LTI Content Selection

### PR DESCRIPTION
**Pull request by thepieterdc**
This PR adds LTI Deep Linking support with Content Selection.

**Tests:**
- Moodle (https://sandbox.moodledemo.net) and local instance
- Brightspace via Ufora (https://ufora.ugent.be); can only be done by deploying to naos.

**Roadmap:**
- [x] Select 1 activity on Moodle
- [x] Select 1 activity on Brightspace
- [x] Improve layout for content selection (#2231)
- [x] Allow multiple items to be selected (#2231)
- [x] Fix cookies (see below)
- [ ] Decide if we want analytics in the iframes (https://github.com/dodona-edu/dodona/pull/2214#discussion_r486623107)

See #2214 for the UI when logging in for the first time.

**Remarks:**
We need to verify that the SameSite policy of our cookies is correct. Because of an update in Chrome, the security for cookies in iframes has been tightened and this might break content selection. Can be temporarily disabled in `chrome://flags` by disabling everything related to SameSite cookies.

Grouping issue: #737

### CODE REVIEW

File: app/controllers/auth/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -5,10 +5,15 @@ class Auth::OmniauthCallbacksController < Devise::OmniauthCallbacksController
...
-    redirect_to root_path
+    # Find the error message and log it for analysis.
+    error_message = failure_message ||
+                    request.params['error_description'] ||
+                    I18n.t('devise.omniauth_callbacks.unknown_failure')
+    logger.error error_message
+
+    # Show a flash message and redirect.
+    flash_failure error_message
+    redirect_to root_path, status: :bad_request
```
Comment (niknetniko) [inline app/controllers/auth/omniauth_callbacks_controller.rb:16]: @thepieterdc What does the `status: :bad_request` do? I ask since it breaks the tests

**Review by charvp — CHANGES_REQUESTED** [review]
There's quite a lot of duplication between the top-level layout erb files. Can we extract some of that to partials?

Otherwise looks good to me, but I'm not familiar with the LTI standard, so I didn't really check for implementation mistakes there. I assume that since things are working on naos, we're good in that department though.

**Comment by niknetniko** [comment]
> There's quite a lot of duplication between the top-level layout erb files. Can we extract some of that to partials?

I intended to rename "embedded" to "lti", but somehow ended up with both. I've removed the "embedded" one (they were identical).

**Comment by charvp** [comment]
> > There's quite a lot of duplication between the top-level layout erb files. Can we extract some of that to partials?
> 
> I intended to rename "embedded" to "lti", but somehow ended up with both. I've removed the "embedded" one (they were identical).

I do think there's still some duplication between `application.html.erb` and `lti.html.erb` (and maybe `frame.html.erb`). Can you look into removing some of that? (It might not be worth it, I'll leave that up to your judgment.)

**Comment by niknetniko** [comment]
> I do think there's still some duplication between `application.html.erb` and `lti.html.erb` (and maybe `frame.html.erb`). Can you look into removing some of that? (It might not be worth it, I'll leave that up to your judgment.)

I've extracted some more common code into partials. I think the remaining code is either different or too small (a few lines) to extract into a partial.

**Review by charvp — APPROVED** [review]

**Review by bmesuere — APPROVED** [review]

**Pull request merged**